Add libplacebo-demos

### Package name

libplacebo-demos

### Brief description of package

libplacebo, but the demo applications only. Main libplacebo package is a dependency of ffmpeg. We cannot build demos directly which depends on ffmpeg, creating dependency cycle as a result.

What I propose is to add libplacebo-demos package with only the demo applications.

Motivation is to give users ability to play around with those applications. Without forcing them to build locally, see https://github.com/haasn/libplacebo/discussions/231

### URL for package's homepage

https://code.videolan.org/videolan/libplacebo

### Provide a basic test case to validate the package's functionality.

_No response_

### MINGW environments where you need the package

- [X] MINGW64
- [X] MINGW32
- [X] UCRT64
- [X] CLANG64
- [X] CLANG32
- [X] CLANGARM64

### Are you willing to submit a PR?

_No response_

It may be possible to add the demos. First, the libplacebo source need to be converted to git repository in PKGBUILD. Otherwise, the demos/3rdparty/nuklear does not exist in tarball and I got the linking error undefined reference to parse_args.

--- a/mingw-w64-libplacebo-demos/PKGBUILD
+++ b/mingw-w64-libplacebo-demos/PKGBUILD
@@ -1,0 +1,60 @@
+# Maintainer: Alexey Pavlov <alexpux@gmail.com>
+
+# Separate package to prevent circular dependency between libplacebo and ffmpeg
+_realname=libplacebo
+pkgbase=mingw-w64-${_realname}-demos
+pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}-demos"
+pkgver=6.338.1
+pkgrel=1
+pkgdesc="Demo programs from libplacebo (mingw-w64)"
+arch=('any')
+mingw_arch=('mingw64' 'ucrt64' 'clang64' 'clangarm64')
+msys2_references=(
+  'archlinux: libplacebo'
+)
+msys2_repository_url='https://code.videolan.org/videolan/libplacebo'
+url='https://libplacebo.org/'
+depends=("${MINGW_PACKAGE_PREFIX}-ffmpeg"
+         "${MINGW_PACKAGE_PREFIX}-libplacebo"
+         "${MINGW_PACKAGE_PREFIX}-SDL2")
+makedepends=("git"
+             "${MINGW_PACKAGE_PREFIX}-cc"
+             "${MINGW_PACKAGE_PREFIX}-meson"
+             "${MINGW_PACKAGE_PREFIX}-pkgconf"
+             "${MINGW_PACKAGE_PREFIX}-python-glad"
+             "${MINGW_PACKAGE_PREFIX}-python-jinja"
+             "${MINGW_PACKAGE_PREFIX}-fast_float"
+             "${MINGW_PACKAGE_PREFIX}-xxhash"
+             "${MINGW_PACKAGE_PREFIX}-vulkan-headers")
+license=('spdx:LGPL-2.1-or-later')
+_commit="2805a0d01c029084ab36bf5d0e3c8742012a0b27"
+source=("${_realname}"::"git+https://code.videolan.org/videolan/libplacebo.git#commit=${_commit}")
+sha256sums=('SKIP')
+
+prepare() {
+  cd "${_realname}"
+  git submodule update --init --depth=1 demos/3rdparty/nuklear
+}
+
+build() {
+  MSYS2_ARG_CONV_EXCL="--prefix=" \
+    ${MINGW_PREFIX}/bin/meson.exe setup \
+    --prefix="${MINGW_PREFIX}" \
+    --buildtype plain \
+    -Ddemos=true \
+    -Dtests=false \
+    -Dd3d11=enabled \
+    -Dlcms=enabled \
+    -Dlibdovi=enabled \
+    -Dshaderc=enabled \
+    -Dvulkan=enabled \
+    "build-${MSYSTEM}" \
+    "${_realname}"
+
+  ${MINGW_PREFIX}/bin/meson.exe compile -C "build-${MSYSTEM}"
+}
+
+package() {
+  install -Dm755 "build-${MSYSTEM}"/demos/*.exe -t "${pkgdir}${MINGW_PREFIX}"/bin/
+  install -Dm644 "${_realname}/LICENSE" "${pkgdir}${MINGW_PREFIX}/share/licenses/${_realname}-demos/LICENSE"
+}

